app/main.py

```diff
@@ -1,2 +1,116 @@
+from typing import Any, Dict, Optional, List
+
+
+class Node:
+    def __init__(self, key: Any, value: Any) -> None:
+        self.key = key
+        self.value = value
+        self.next = None
+
+
 class Dictionary:
-    pass
+    def __init__(self, capacity: int = 8, load_factor: float = 0.7) -> None:
+        self.capacity = capacity
+        self.load_factor = load_factor
+        self.size = 0
+        self.table: List[Optional[Node]] = [None] * self.capacity
+
+    def hash_function(self, key: Any) -> int:
+        return hash(key) % self.capacity
+
+    def __setitem__(self, key: Any, value: Any) -> None:
+        index = self.hash_function(key)
+        new_node = Node(key, value)
+
+        if self.table[index] is None:
+            self.table[index] = new_node
+        else:
+            current = self.table[index]
+            while current:
+                if current.key == key:
+                    current.value = value
+                    return
+                current = current.next
+            new_node.next = self.table[index]
+            self.table[index] = new_node
+
+        self.size += 1
+        if self.size >= self.load_factor * self.capacity:
+            self.resize()
+
+    def __getitem__(self, key: Any) -> Any:
+        index = self.hash_function(key)
+        current = self.table[index]
+        while current:
+            if current.key == key:
+                return current.value
+            current = current.next
+        raise KeyError(f"Key '{key}' not found")
+
+    def __len__(self) -> int:
+        return self.size
+
+    def __delitem__(self, key: Any) -> None:
+        index = self.hash_function(key)
+        current = self.table[index]
+        prev = None
+        while current:
+            if current.key == key:
+                if prev:
+                    prev.next = current.next
+                else:
+                    self.table[index] = current.next
+                self.size -= 1
+                return
+            prev = current
+            current = current.next
+        raise KeyError(f"Key '{key}' not found")
+
+    def clear(self) -> None:
+        self.capacity = 8
+        self.size = 0
+        self.table = [None] * self.capacity
+
+    def get(self, key: Any, default: Any = None) -> Any:
+        try:
+            return self[key]
+        except KeyError:
+            return default
+
+    def pop(self, key: Any, default: Any = None) -> Any:
```

Comment:
FYI:
d = {}
d.pop(1) # Raise KeyError
d.pop(1, None) # not raise
Implement it into pop method